Doctrine/UserManager.php

```diff
@@ -48,10 +43,15 @@ public function __construct(PasswordUpdaterInterface $passwordUpdater, Canonical
         parent::__construct($passwordUpdater, $canonicalFieldsUpdater);
 
         $this->objectManager = $om;
-        $this->repository = $om->getRepository($class);
+        $this->class = $class;
+    }
 
-        $metadata = $om->getClassMetadata($class);
-        $this->class = $metadata->getName();
+    /**
+     * @return ObjectRepository
+     */
+    protected function getRepository()
```

Comment:
We have many users extending the class currently. So this would be a hard BC break (I'm in favor of deprecating the ability to extend it in the future though, but not about making it private hardly just before the 2.0 stable release)